Update HPC OnDemand instructions, beautify some things, add bitterroot, civet-like environment how-to

milljm opened the pull request:
Add information about not using the visualization machines for build MOOSE/MOOSE-based applications (the real reason for this PR).

Add moosetainer (or possibly apptainer) instructions.

Modernize the HPC OnDemand page, which was making use of these instructions.

~~This is a work in progress.~~

Closes #27731

Commit:
WIP: Add `moosetainer` (Apptainer) instructions

Add information about not using the visualization machines for
build MOOSE/MOOSE-based applications (the real reason for this
PR).

Add moosetainer (or possibly apptainer) instructions.

Modernize the HPC OnDemand page, which was making use of these
instructions.

This is a work in progress.

Closes #27731

Commit:
WIP awesomeness

cticenhour commented on framework/doc/acronyms.yml:
```diff
@@ -63,3 +63,4 @@ FAR: Failure Analysis Report
 QLD: Quality Level Determination
 QP: quadrature point
 QPs: quadrature points
+URI: Uniform resource identifier
```
```suggestion
URI: uniform resource identifier
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -2,9 +2,12 @@
-The MOOSE build system is a hierarchical Makefile based system.  The idea is that no matter where you
-are in the hierarchy of MOOSE applications/modules/framework, simply typing `make` will rebuild
-everything that needs to be rebuilt.  The basic way to build any MOOSE application is simply to do a:
+!style! halign=left
+The MOOSE build system is a hierarchical Makefile based system. The idea is that no matter where
+you are in the hierarchy of MOOSE applications/modules/framework, simply typing `make` will rebuild
+everything that needs to be rebuilt. The basic way to build any MOOSE application is simply to do
+a:
+!style-end!
```
FYI - if you wanted to reduce new line count, simply placing `!style halign=left` in front of the paragraph and not the full "start-to-end" syntax is sufficient to get the spacing right after a heading. Mentioning it in case you wanted to adjust as you're working through other parts of this review, but I will not require it, as this is good syntax as well.

milljm replied:
Oh! I didn't know that. Yes I'd much rather not need the style-end syntax.

milljm commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -129,16 +142,22 @@ Make sure to prepend your directories with a `%` sign.
+The Revision file contains both a version string (git hash) and a revision string containing more
+detailed information or tagged information. Note that the name of your app will appear instead of
+"MOOSE" for your individual application:
+
```
lol, this was never rendering properly.

ref: https://mooseframework.inl.gov/application_development/build_system.html#f257438f-88ba-4daf-887e-8e03cce37149

cticenhour replied:
Those empty lines are super easy to forget....

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -44,80 +49,88 @@ some of the environment variables you can set and their effects:
-`METHODS`.
+`METHODS` is used to specify the full set (space separated) of the above options that you will
+ _ever_ want to used.  It is used in the build of `libMesh` to build a `libMesh` library for each of
+ the `METHODS`.
```
```suggestion
 the `METHODS`, so that a rebuild of `libMesh` is not required every time a rebuild of MOOSE 
 with a new `METHOD` is performed.
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -44,80 +49,88 @@ some of the environment variables you can set and their effects:
-## Build System Optimization
+## Build Optimization
 
+!style! halign=left
```
```suggestion
!alert! note
```

Let's make this stand out a bit more since the section isn't strictly necessary.

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -44,80 +49,88 @@ some of the environment variables you can set and their effects:
 This is advanced information meant to inform you of optimizations that happen within the MOOSE build
-system.  This is not necessary to understand how to use and build MOOSE.
+system. This is not necessary to understand how to use and build MOOSE.
+!style-end!
```
```suggestion
!alert-end!
```

Tied to my previous comment.

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -44,80 +49,88 @@ some of the environment variables you can set and their effects:
 ### Header Symlinking
 
+!style! halign=left
 MOOSE and MOOSE-based application's include directories are organized into sub-directories for each
```
```suggestion
MOOSE and MOOSE-based applications include directories are organized into sub-directories for each
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -44,80 +49,88 @@ some of the environment variables you can set and their effects:
 systems, meaning that each application could potentially have 30+ directories to search for include
-files.  In addition, applications can compose/use each other and use the physics modules.  Each time
+files. In addition, applications can compose/use each other and use the physics modules. Each time
 you add another application/module you're gaining another 30+ include directories.
```
```suggestion
you add another application/module, you're gaining another 30+ include directories.
```

milljm commented on modules/doc/content/getting_started/installation/build_petsc_and_libmesh.md.template:
```diff
@@ -1,16 +1,22 @@
-./update_and_rebuild_wasp.sh {{WASP_ARGS1}}
+./update_and_rebuild_petsc.sh {{PETSC_ARGS1}} {{PETSC_ARGS2}} || return
+./update_and_rebuild_libmesh.sh {{LIBMESH_ARGS1}} || return
+./update_and_rebuild_wasp.sh {{WASP_ARGS1}} || return
```
This should prevent folks who copy paste the whole code block instead of one line at time, when returning to the Discussion forum with errors beyond the first (petsc fails, yet they seem to be running update_and_rebuild_libmesh.sh)

GiudGiud replied:
pretty cool

milljm replied:
ah! thanks for commenting. Reminds me to go and look for: I think I have ` || exit 1` elsewhere, and that is wrong... it should be ` || return` like this change.

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -44,80 +49,88 @@ some of the environment variables you can set and their effects:
-sets of applications/modules/framework on the Falcon computer at Idaho National Laboratory.  It is
-plain to see how much time is saved by using both header symlinking and unity builds.
+provide a huge savings in compile time. The following is showing timing for `make -j 16` for
+various sets of applications/modules/framework on the Falcon computer at Idaho National Laboratory.
```
```suggestion
various sets of applications/modules/framework on the Falcon supercomputer at Idaho National Laboratory.
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -44,80 +49,88 @@ some of the environment variables you can set and their effects:
 into the final library / application.
+!style-end!
 
 A different idea is to compile all of these files together, providing considerable savings for
```
```suggestion
A different idea is to compile all of these files together as one file, providing considerable savings for
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -44,80 +49,88 @@ some of the environment variables you can set and their effects:
+!style-end!
 
 A different idea is to compile all of these files together, providing considerable savings for
 evaluating the `#include`s and ultimately just producing _one_ `.o` file to be linked.  This can
```
```suggestion
evaluating the `#include`s and ultimately just producing _one_ `.o` file to be linked.  This is known as a "unity build" and can
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -44,80 +49,88 @@ some of the environment variables you can set and their effects:
 
-The way we achieve this in MOOSE is using the idea of "Unity Builds".  A Unity Build compiles
-multiple `.C` files together by create a `.C` file that `#include`s all of the other `.C` files.  In
+The way we achieve this in MOOSE is using the idea of "Unity Builds". A Unity Build compiles
+multiple `.C` files together by create a `.C` file that `#include`s all of the other `.C` files. In
```
```suggestion
The way we achieve a unity build in MOOSE is by compiling
multiple `.C` files together by create a `.C` file that `#include`s all of the other `.C` files. In
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -44,80 +49,88 @@ some of the environment variables you can set and their effects:
-default `src` and `src/base` are not unity built because they generally contain a mix of objects that
-don't benefit from being built simultaneously.  To add to that list, set the `app_non_unity_dirs`
-variable in your `Makefile` before the `include` of `app.mk` like so:
+In addition you can also keep from unity building individual directories within your project. By
```
```suggestion
In addition, you can also keep from using a unity build for individual directories within your project. By
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -44,80 +49,88 @@ some of the environment variables you can set and their effects:
-don't benefit from being built simultaneously.  To add to that list, set the `app_non_unity_dirs`
-variable in your `Makefile` before the `include` of `app.mk` like so:
+In addition you can also keep from unity building individual directories within your project. By
+default `src` and `src/base` are not unity built because they generally contain a mix of objects
```
```suggestion
default `src` and `src/base` are not built this way because they generally contain a mix of objects
```

Reducing awkward phrasing since what "this" is is clear from the previous sentence.

milljm commented on modules/doc/content/help/faq/apptainer.md:
```diff
@@ -0,0 +1,160 @@
+Beyond the initial cloning of the application, most steps are barred from network access. You
+can mimmic these behaviors if need be. See [Apptainer Features](help/faq/apptainer.md#features)
+below for details.
+!alert-end!
```
TODO: I need to add how to enable this feature.

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -129,16 +142,22 @@ Make sure to prepend your directories with a `%` sign.
+!style! halign=left
+The build process can generate C++ header files containing repository versioning information for use
+within your application. These header files are generated by default but can be disabled by setting
+a Make variable in your application's Makefile:
```
```suggestion
a `Make` variable in your application's `Makefile`:
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -151,31 +170,37 @@ name of your app will appear instead of "MOOSE" for your individual application:
-as the run-time experience and that philosophy shows in the way we've structured the Makefiles for building
-MOOSE-based applications.
+!style! halign=left
+MOOSE uses the tried and true Unix standard build tools,
```
```suggestion
MOOSE uses the tried and true UNIX standard build tools,
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -151,31 +170,37 @@ name of your app will appear instead of "MOOSE" for your individual application:
+MOOSE uses the tried and true Unix standard build tools,
+[GNU Make](https://www.gnu.org/software/make/) and
+[GNU Autoconf](https://www.gnu.org/software/autoconf/). These tools are ubiquitous, reliable, and
+stable, where many other build systems are not. We believe that usability of tool includes the build
```
```suggestion
stable, where many other build systems are not. We believe that usability of a tool includes the build
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -185,92 +210,112 @@ You'll probably want to add the path above to your PATH.
-- `framework/configure.ac` - This is the input file that is processed by "autoconf" to create the configure
-  script. If you modify this file you will need to rerun `autoconf` to regenerate the `configure` script.
-  If you defined a new variable (e.g. with AC_DEFINE), you will need to run `autoheader` then `autoconf`.
+- `framework/configure.ac` - This is the input file that is processed by "autoconf" to create the
```
```suggestion
- `framework/configure.ac` - This is the input file that is processed by `autoconf` to create the
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
````diff
@@ -185,92 +210,112 @@ You'll probably want to add the path above to your PATH.
 ```
 
-Additionally each application may wish to make an one or more file structures available for end users to install into their own environment. This allows for end-users to install and run the application test suite, examples, tutorials, or other inputs of interest. To customize what is available for installation, the application developer will define the list of source directories that can be insta ...
+Additionally each application may wish to make an one or more file structures available for end
+users to install into their own environment. This allows for end-users to install and run the
````
```suggestion
Additionally each application may wish to make an one or more file structures available for end-users
to install into their own environment. This allows for end-users to install and run the
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
````diff
@@ -185,92 +210,112 @@ You'll probably want to add the path above to your PATH.
 tests  example_01  example_02  tutorials
 ```
 
 !alert warning
````
```suggestion
!alert! warning
```
Since we are using two lines after this alert, let's use the full syntax to ensure it is rendered properly

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -185,92 +210,112 @@ You'll probably want to add the path above to your PATH.
 !alert warning
-Make sure that installed paths do not include source code. This could result in an accidental export of information for those cleared for only binary access.
+Make sure that installed paths do not include source code. This could result in an accidental export
+of information for those cleared for only binary access.
```
```suggestion
of information for those cleared for only binary access.
!alert-end!
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -185,92 +210,112 @@ You'll probably want to add the path above to your PATH.
-Once an application has been installed, end users that have access to those binaries can install or copy those paths to their home directories and also run the tests using the test harness using convenient command line parameters.
+!style! halign=left
+Once an application has been installed, end users that have access to those binaries can install or
+copy those paths to their home directories and also run the tests using the test harness using
```
```suggestion
copy those paths to their home directories and also run the tests with the MOOSE test harness using
```

cticenhour commented on modules/doc/content/getting_started/installation/build_petsc_and_libmesh.md.template:
```diff
@@ -1,16 +1,22 @@
 !alert! tip
 `MOOSE_JOBS` is a loose influential environment variable that dictates how many cores to use when
-executing many of our scripts.
+executing many of our scripts. While operating on [!ac](INL) [!ac](HPC) login nodes along side
```
```suggestion
executing many of our scripts. While operating on [!ac](INL) [!ac](HPC) login nodes alongside
```

cticenhour commented on modules/doc/content/getting_started/installation/build_petsc_and_libmesh.md.template:
```diff
@@ -1,16 +1,22 @@
 `MOOSE_JOBS` is a loose influential environment variable that dictates how many cores to use when
-executing many of our scripts.
+executing many of our scripts. While operating on [!ac](INL) [!ac](HPC) login nodes along side
+everyone else, it is courtesy to limit your CPU Core usage. We prefer folks limit themselves to 6:
```
```suggestion
everyone else, it is courtesy to limit your CPU core usage. We prefer that users limit themselves to 6:
```

cticenhour commented on modules/doc/content/getting_started/installation/inl_hpc_install_moose.md:
```diff
@@ -3,24 +3,35 @@
 
 Requesting access (shell account) to [!ac](INL) [!ac](HPC) machines is handled by the
-[NCRC](https://inl.gov/ncrc/) group.
+[NCRC](https://inl.gov/ncrc/) group. Once access has been granted, one can use
```
```suggestion
[INL Nuclear Computational Resource Center](https://inl.gov/ncrc/). Once access has been granted, one can use
```

cticenhour commented on modules/doc/content/getting_started/installation/inl_hpc_install_moose.md:
```diff
@@ -3,24 +3,35 @@
 Requesting access (shell account) to [!ac](INL) [!ac](HPC) machines is handled by the
-[NCRC](https://inl.gov/ncrc/) group.
+[NCRC](https://inl.gov/ncrc/) group. Once access has been granted, one can use
+[HPC Ondemand](hpc_ondemand.md) services to gain access to a shell.
```
```suggestion
[HPC OnDemand](hpc_ondemand.md) services to gain access to a shell.
```

cticenhour commented on modules/doc/content/getting_started/installation/inl_hpc_install_moose.md:
```diff
@@ -3,24 +3,35 @@
 document), please head over to [NCRC Applications](help/inl/applications.md) instead, and choose the
 application applicable to you.
 
+At the time of this writing the following [!ac](INL) [!ac](HPC) clusters are available for use:
```
```suggestion
At the time of this writing, the following [!ac](INL) [!ac](HPC) clusters are available for use:
```

cticenhour commented on modules/doc/content/getting_started/installation/inl_hpc_install_moose.md:
```diff
@@ -32,15 +43,15 @@ echo "module load use.moose moose-dev" >> ~/.bash_profile
-Follow the below instructions, and replace `cluster_name` placeholder with either `lemhi` or
-`sawtooth` accordingly.
+Follow the below instructions, and replace all occurrences of `cluster_name` with either `lemhi`,
+`sawtooth`, or `bitterroot` accordingly with the machine you chose to operate on.
```
```suggestion
`sawtooth`, or `bitterroot` accordingly with the machine you choose to operate on.
```

cticenhour commented on modules/doc/content/getting_started/new_users.md.template:
```diff
@@ -109,12 +109,14 @@ you to visualize and process the results of your simulations.
 
 - `--prefix`: Specify a target installation path. Useful if you plan on running `make install`.
 
+- See [MOOSE Build System](application_development/build_system.md) for in-depth information.
```
```suggestion
- See [MOOSE Build System](application_development/build_system.md) documentation for in-depth information.
```

cticenhour commented on modules/doc/content/ncrc/hpc_cluster_information.md:
```diff
@@ -1,5 +1,7 @@
-|----------|--------------------|--------------|
-| Lemhi    | 40                 | 72           |
-| Sawtooth | 48                 | 168          |
+Additional information for each cluster can be found at [INL HPC](https://inl.gov/hpc).
```
```suggestion
Additional information for each cluster can be found at the [INL HPC website](https://inl.gov/hpc).
```

cticenhour commented on modules/doc/content/ncrc/applications/ncrc_develop.md.template:
```diff
@@ -74,10 +82,10 @@ cd ~/projects/{{binary}}
+The `fetch upstream` git command updates your local references with that of the upstream remote. It
+does not modify any source files. If nothing returns, then there are no upstream updates. If you
+have built {{ApplicationName}} once before, you are finished and can safely exit this documentation.
+First time users will of course need to continue.
```
```suggestion
First-time users will of course need to continue.
```

cticenhour commented on modules/doc/content/help/inl/hpc_ondemand.md:
```diff
@@ -1,96 +1,55 @@
 # HPC OnDemand
 
-[HPC OnDemand](https://hpcondemand.inl.gov/pun/sys/dashboard), is a service provided by the INL, which allows a user direct access to the resources contained within the HPC enclave via their web browser. In order to utilize this service, you must first [request an account](https://inl.gov/ncrc).
+[HPC OnDemand](https://hpcondemand.inl.gov/pun/sys/dashboard), is a service provided by the INL,
```
```suggestion
[HPC OnDemand](https://hpcondemand.inl.gov/pun/sys/dashboard), is a service provided by INL
```

cticenhour commented on modules/doc/content/help/inl/hpc_ondemand.md:
```diff
@@ -1,96 +1,55 @@
-Using moosebuild (or rather Singularity) in this fashion, limits you to only having the resources available on that one physical node. We are looking into how we might allow PBS to launch Singularity containers (wouldn't that be cool), but this feature is some ways out.
+- Interactive Apps, Linux Desktop / Linux Deskop with Visualization
 
+  - Launches a new browser tab allowing you to schedule an interactive session to one of
```
```suggestion
  - Launches a new browser tab allowing you to schedule an interactive session on one of
```

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
````diff
@@ -0,0 +1,160 @@
+```
+
+With the [!ac](URI) known, launch an interactive shell using
+[HPC Ondemand](hpc_ondemand.md#interactive-shell-idinteractive-shell) for any of the [!ac](INL)
````
```suggestion
[HPC OnDemand](hpc_ondemand.md#interactive-shell-idinteractive-shell) for any of the [!ac](INL)
```

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
```diff
@@ -0,0 +1,160 @@
+## Troubleshoot
+
+!style! halign=left
+You are now operating from within the same container that which your PR is failing. By default,
```
```suggestion
You are now operating from within the same container within which your PR is failing. By default,
```

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
```diff
@@ -0,0 +1,160 @@
+your project.
+!style-end!
+
+Next, simply put; Clone your application (if you have not). Build it. Run it. Test it. Do whatever
```
```suggestion
Next, simply put: clone your application (if you have not); build it; run it; and test it. Do whatever
```

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
```diff
@@ -0,0 +1,160 @@
+# Civet-like Environment
```
"CIVET" instead? If so, this will need to be corrected several times. Noting once here.

milljm replied:
Actually I was thinking of changing this to Apptainer Environment.

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
```diff
@@ -0,0 +1,160 @@
+  been developing in.
+
+- Re-visit the failing step occurring in Civet, and scrutinize with the utmost care all commands,
+  arguments used, environment variables set, etc. Even going so far as checking the steps that
```
```suggestion
  arguments used, environment variables set, etc. One should even go so far as checking the steps that
```

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
```diff
@@ -0,0 +1,160 @@
+
+If you're still unable to reproduce the error, please keep in mind one golden truth:
++Containers are immutable+. The lack of reproducibility is being caused by *something* that is
+different in your environment compared to Civet's. Things like Network connectivity, shell flavor
```
```suggestion
different in your environment compared to Civet's. Things like network connectivity, shell flavor
```

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
```diff
@@ -0,0 +1,160 @@
+If you're still unable to reproduce the error, please keep in mind one golden truth:
++Containers are immutable+. The lack of reproducibility is being caused by *something* that is
+different in your environment compared to Civet's. Things like Network connectivity, shell flavor
+(tsch, zsh, bash), CPU microachitechture (extremely rare), and the like.
```
```suggestion
(tsch, zsh, bash), CPU microarchitecture (extremely rare), and the like.
```

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
```diff
@@ -0,0 +1,160 @@
+#### Paths and Mounts
+
+!style! halign=left
+You can instruct Apptainer to mount any path you have access to, to be made available inside the
```
```suggestion
You can instruct Apptainer to mount any path you have access to, so that it is available inside the
```

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
````diff
@@ -0,0 +1,160 @@
+[moose-dev][~]> ls /scratch  # contents of scratch is displayed
+```
+
+Do you want to mount something somewhere else? Also with read-only permissions?
````
```suggestion
Do you want to mount something somewhere else - perhaps also with read-only permissions?
```

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
````diff
@@ -0,0 +1,160 @@
+```
+
+!alert! tip title=Custom Prompt
+There is an all encompasing `$CUSTOM_PROMPT` variable that allows you to pass your own prompt:
````
```suggestion
There is an all-encompassing `$CUSTOM_PROMPT` variable that allows you to pass your own prompt:
```

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
````diff
@@ -0,0 +1,160 @@
+[my-container][12:17:43]>
+```
+
+<!-- NOTE to editor: sub children elements require less top possitioning (-15 vs -7) -->
````
```suggestion
<!-- NOTE to editor: sub children elements require less top positioning (-15 vs -7) -->
```

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2301347/) on 6a6c758 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/27744/site)

This comment will be updated on new commits.

Commit:
Address comments

Fix syntax, spelling. Other small adjustments.

Adress TODO: Features (no network, and other useful arguments).

Commit:
Convert to table

Convert the helpful argument syntax to a table

Commit:
Remove apptainer.md from INL/ help

There are other applications which only are including a small subset
of /inl/ documentation (Bison). This change is to simply address the
failing build of Bison documentation.

Commit:
Update heat_conduction to heat_transfer

Example used was outdated.

Commit:
address syntax/grammar

Commit:
More grammar and syntax adjustments

Commit:
more syntax / grammar changes

Commit:
more syntax / grammar changes

grmnptr commented on modules/doc/content/help/faq/apptainer.md:
```diff
@@ -0,0 +1,174 @@
+# Apptainer
```
I think we should move this to something like:
```
Getting started->Installation->Apptainer
```
We already have an option for docker here:
https://mooseframework.inl.gov/getting_started/installation/index.html

We could have the same with apptainer.
What do you think?

milljm replied:
Maybe, but it is only useful for those with access to INL HPC machines. I wanted to group this in with that somehow.

cticenhour replied:
Would it be better to place this in `moose/doc/content/help/inl` then, in order to be directly beside the other HPC-related documents?

milljm replied:
I originally put it there, but that breaks Bison documentation.

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -67,26 +64,25 @@ as a suffix like: `something-$METHOD`. The valid options are:
 `METHODS` is used to specify the full set (space separated) of the above options that you will
  _ever_ want to used.  It is used in the build of `libMesh` to build a `libMesh` library for each of
- the `METHODS`.
+ the `METHODS`, so that a rebuild of `libMesh` is not required every time a rebuild of MOOSE.
```
```suggestion
 the `METHODS`, so that a rebuild of `libMesh` is not required for every rebuild of MOOSE.
```

cticenhour commented on modules/doc/content/application_development/build_system.md:
```diff
@@ -99,37 +95,37 @@ that directory. This means that each application/module you add only adds _one_
-The way we achieve this in MOOSE is using the idea of "Unity Builds". A Unity Build compiles
-multiple `.C` files together by create a `.C` file that `#include`s all of the other `.C` files. In
-MOOSE this happens in a directory called `build/unity_src` in each application / module.
+The way we achieve a unity build in MOOSE is by compiling multiple `.C` files together by create a
```
```suggestion
The way we achieve a unity build in MOOSE is by compiling multiple `.C` files together by creating a
```

cticenhour commented on modules/doc/content/help/faq/apptainer.md:
```diff
@@ -43,63 +42,66 @@ lengthy prompt)
-  does: Clone your project accordingly just as the `Fetch and Branch` step. Don't use the one you
-  been developing in.
+- Be absolutely certain that your project is clean. It might even be best to do exactly what
+  [!ac](CIVET) does: Clone your project accordingly just as the `Fetch and Branch` step. Don't use
+  the one you been developing in.
```
```suggestion
  [!ac](CIVET) does: Clone a clean copy of your project just as in the `Fetch and Branch` step. Don't use
  the one you have been developing in.
```

milljm replied:
I am not sure I like this change. A clean copy to a lot of users is "cloning it". But Civet may be excluding, or including submodules. Which is why I prefer "clone how Civet clones". Something along those lines.

Do you have another suggesting other than removing "accordingly"? Maybe "identically"?

milljm replied:
@cticenhour Just pinging on you this, if you're okay with the new change.

cticenhour commented on modules/doc/content/help/inl/hpc_ondemand.md:
```diff
@@ -42,8 +41,7 @@ The following represents some of the more useful features, and is not an exhaust
    selected. The interactive shell will be one of the more useful tools when needing direct access
-   to things like: launching PBS jobs manually, building MOOSE, obtaining a
-   [Civet-like testing environment](faq/apptainer.md) and more.
+   to things like: launching PBS jobs manually, building MOOSE, running tests, and more.
```
```suggestion
   to things like: launching PBS/Slurm jobs manually, building MOOSE, running tests, and more.
```

Since Slurm is now a thing at INL on the newest machine. 😄

cticenhour reviewed (dismissed):
A few more minor comments.

Commit:
Grammar/Syntax changes, move apptainer to help/inl

Commit:
re-reword 'accordingly' for repo cloning hints

loganharbour committed:
Update HPC install to use container modules

milljm commented:
I forgot about this PR. I need/want to work on it... I think we neglected to realize the current Install Instructions are now sorta... well, not complete. If you click on "Getting Started, Install MOOSE, INL's HPC Cluster", we basically jump right into running `./versioner.py`, without explaining how to clone moose... I feel we skipped a whole page somehow.

Commit:
WIP: commit to work on another machine

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2509704/), step Docs: sync website on 8971ab3 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/27744/site)

This comment will be updated on new commits.

Commit:
simplifications

Commit:
TLC HPC instructions

Modified the index site to bring INL HPC into the top portion of the
list. This is seeing enough traffic now, I think it belongs up here.

Include instructions on how to clone MOOSE (or ask the user to clone
their application at the very least).

Commit:
Bah, forgot to add the recap

Add a recap section to demonstrate how easy it is, once you get used
to it.

Closes #27731

Commit:
Drop a level of heading

Seems to be a bug in MooseDocs with level 4 headings. When used:

It creats an out-of-order TOC as described above.

GiudGiud commented on modules/doc/content/application_development/build_system.md:
````diff
@@ -185,92 +203,109 @@ You'll probably want to add the path above to your PATH.
+target will copy the binary, and required libraries to a file tree based on the prefix you
+configured with (set by the variable `PREFIX` or the `--prefix` argument).
 
 ```
````
```suggestion
```bash
```

Commit:
adjust counts, fix syntax/grammar/spelling

address requested changes (grammar/spelling).
Adjust counts to how many build systems MOOSE contains.

loganharbour commented on modules/doc/content/getting_started/installation/inl_hpc_install_moose.md:
````diff
@@ -4,32 +4,66 @@ The following instructions are for those operating on [!ac](INL) [!ac](HPC) mach
+assumes you cloned MOOSE as directed in the previous step):
 
 ```bash
+cd /scratch/<your INL HPC User ID>/projects/moose
````
People still can't do this on most hosts due to missing packages; we should probably add in a component about how to get a working env first. Because this won't work on its own

loganharbour commented on modules/doc/content/help/inl/apptainer.md:
````diff
@@ -0,0 +1,177 @@
+
+```language=yaml
+//: Running in versioned apptainer container moose-dev
+BUILD_ROOT/moose/: Executing moosebuild in environment oras://mooseharbor.hpc.inl.gov/moose-dev/moose-dev-x86_64:4b79189
````
Is this from a really old PR? We don't build moose-dev-x86_64 anymore

loganharbour commented on modules/doc/content/help/inl/apptainer.md:
```diff
@@ -0,0 +1,177 @@
+[~]> module load apptainer
+[~]> apptainer shell oras://mooseharbor.hpc.inl.gov/moose-dev/moose-dev-x86_64:4b79189
+
+[sha256:52660332a1977bb9cbb3e4aaf9b19f810669eba8579b8dda6193eba7b05cf359][~]>
```
We should use a newer example because the prompt is much more rich now

loganharbour commented on modules/doc/content/help/inl/apptainer.md:
````diff
@@ -0,0 +1,177 @@
+displayed:
+
+```bash
+[~]> export APPTAINERENV_APPTAINER_NAME=moose-dev
````
I don't believe this will work anymore with our new prompt

loganharbour reviewed (commented):
woo, review that I never hit the green button for
